python/mxnet/gluon/contrib/estimator/estimator.py

```diff
@@ -0,0 +1,376 @@
+            data, label = self._get_data_and_label(batch, self.context)
+            pred = [self.net(x) for x in data]
+            loss = [self.loss[0](y_hat, y) for y_hat, y in zip(pred, label)]
+            # update metrics
+            for metric in val_metrics:
+                if isinstance(metric, Loss):
+                    metric.update(0, loss)
+                else:
+                    metric.update(label, pred)
+
+    def fit(self, train_data,
+            val_data=None,
+            epochs=1,
+            event_handlers=None):
+        """Trains the model on a given dataset for a specified
+        number of epochs. Also, the batch size is inferred from the
+        DataLoader's batch_size.
+
+        Parameters
+        ----------
+        train_data : DataLoader
+            training data with data and labels
+        val_data : DataLoader
+            validation data with data and labels
+        epochs : int, default 1
+            number of epochs to iterate on the training data.
+        batch_size : int
+            number of samples per gradient update.
+            default will be 32 per device
+        event_handlers : EventHandler or list of EventHandler
+            list of EventHandlers to apply during training
+        batch_fn : function
+            custom batch function to extract data and label
+            from a data batch and load into contexts(devices)
+        """
+        self.max_epochs = epochs
+
+        # provide default handlers
+        event_handlers = self._prepare_default_handlers(val_data, event_handlers)
+
+        train_begin, epoch_begin, batch_begin, \
+        batch_end, epoch_end, train_end = self._categorize_handlers(event_handlers)
+
+        # only pass a weak reference to all event handlers
+        estimator_ref = weakref.proxy(self)
+        # training begin
+        for handler in train_begin:
+            handler.train_begin(estimator_ref)
+
+        for epoch in range(epochs):
+            # epoch begin
+            for handler in epoch_begin:
+                handler.epoch_begin(estimator_ref)
+
+            for i, batch in enumerate(train_data):
+                if not isinstance(train_data, gluon.data.DataLoader):
+                    raise ValueError("Estimator only support input as Gluon DataLoader. Alternatively, you "
+                                     "can transform your DataIter or any NDArray into Gluon DataLoader. "
+                                     "Refer to gluon.data.dataloader")
+                data, label = self._get_data_and_label(batch, self.context)
+
+                batch_size = batch[0].shape[0]
+
+                # batch begin
+                for handler in batch_begin:
+                    handler.batch_begin(estimator_ref, batch=batch)
+
+                with autograd.record():
+                    pred = [self.net(x) for x in data]
+                    loss = [self.loss[0](y_hat, y) for y_hat, y in zip(pred, label)]
```

Comment:
same as above, support 1 loss for now, will reopen https://github.com/apache/incubator-mxnet/pull/14628 once this is finalized.